scalafix-tests/unit/src/test/scala/scalafix/tests/reflect/RuleDecoderSuite.scala

```diff
@@ -26,13 +27,42 @@ class RuleDecoderSuite extends AnyFunSuite {
     RuleDecoder.Settings().withCwd(cwd)
   val decoder: ConfDecoder[Rules] = RuleDecoder.decoder(decoderSettings)
   val expectedName = "NoDummy"
-  val expectedDescription = ""
+
   test("absolute path resolves as is", SkipWindows) {
     val rules = decoder.read(Conf.Str(s"file:$abspath")).get
     assert(expectedName == rules.name.value)
   }
+
   test("relative resolves from custom working directory") {
     val rules = decoder.read(Conf.Str(s"file:$relpath")).get
     assert(expectedName == rules.name.value)
   }
+
+  test("resolved classes can be reloaded", SkipWindows) {
```

Comment:
I would keep a separate commit for that particular change instead of merging it to the big unit one, as the reasoning for the change (both for scala 2 & scala3) is quite tricky.